### PR TITLE
NONJAVACLI-3632: Non-java clients identity pool extension comma parsing

### PR DESCRIPTION
**Pull request by PratRanj07**
This PR intends to implement CSV format parsing for strings related to SASL OAuthbearer Extensions. Jira Ticket:  [NONJAVACLI-3632](https://confluentinc.atlassian.net/browse/NONJAVACLI-3632)

### CODE REVIEW

File: src/rdstring.c [hunk 1]
```diff
@@ -553,6 +553,135 @@ char **rd_string_split(const char *input,
...
+ * Supports quoted fields using `"`. Escaped characters (e.g., `\n`, `\t`) are
+ * also handled. Quoted fields can contain separators, and the quotes will be
+ * removed in the output. The array and the array elements will be allocated
+ * together and must be freed with a single rd_free(array) call. The array
+ * elements are copied, and any escape sequences or quotes are removed.
+ *
+ * @returns The parsed fields in an array. The number of elements in the
+ *          array is returned in \p cntp.
+ */
+char **rd_string_split_csv(const char *input,
```
Comment (anchitj) [inline src/rdstring.c:578]: I think we should return error code and also handle the cases where input string is invalid, like `"` started but no ending `"` present.

File: src/rdstring.c [hunk 2]
```diff
@@ -632,6 +761,68 @@ static int ut_string_split(void) {
...
+                {"extension1:extension1val",
+                 "extension2:extension2val1,extension2val2"},
+            },
+            {
+                "extension1:extension1val,extension2:extension2val1",
+                ',',
+                rd_true,
+                2,
+                {"extension1:extension1val", "extension2:extension2val1"},
+            },
```
Comment (anchitj) [inline src/rdstring.c:790]: Can we also add other edge cases like empty string, string with just a single k:v pair, and using escape sequences. And also test for invalid strings.

**Review by anchitj — CHANGES_REQUESTED** [review]

**Comment by anchitj** [comment]
Thanks @PratRanj07. Some minor comments, but overall looks good. Let's verify manually with a cluster too.

File: src/rdstring.c [hunk 3]
```diff
@@ -553,6 +553,142 @@ char **rd_string_split(const char *input,
...
+        char *p;
+        char **arr;
+        size_t inputlen;
+        size_t i            = 0;
+        size_t elen         = 0;
+        rd_bool_t in_quotes = rd_false;
+
+        *cntp = 0;
+
+        // Count approximate field count (ignore sep inside quotes)
```
Comment (anchitj) [inline src/rdstring.c:596]: Please place comments in /* */ blocks

**Comment by anchitj** [comment]
Also verify if the extensions have spaces can they be parsed correctly and interpreted by the broker. Like `identityPoolId=p1,       p2`
